p2p/server/deadline_adjuster_test.go

```diff
@@ -0,0 +1,128 @@
+package server
+
+import (
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/libp2p/go-yamux/v4"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/spacemeshos/go-spacemesh/p2p/server/mocks"
+)
+
+func TestDeadlineAdjuster(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	s := mocks.NewMockpeerStream(ctrl)
+	clock := clockwork.NewFakeClock()
+
+	readChunks := []string{"xy", "ABCD", "EF", "0123", "4567", "89"}
+	writeChunks := []string{"foo", "abcd", "efgh", "ijk", "bbbc"}
```

Comment:
Casts could be avoided with
```suggestion
	readChunks := []byte{"xy", "ABCD", "EF", "0123", "4567", "89"}
	writeChunks := []byte{"foo", "abcd", "efgh", "ijk", "bbbc"}
```

Reply:
You probably meant `[][]byte` instead of `[]byte`?
Anyway, this will require extra casts
```
readChunks := [][]byte{[]byte("xy"), []byte("ABCD"), []byte("EF"), []byte("0123"), []byte("4567"), []byte("89")}
```